src/main.rs

```diff
@@ -57,8 +57,11 @@ fn main() {
     // line) to win the conflict.
     //
     if args.dump.is_some() && args.archive.is_some() {
-        match (m.occurrences_of("dump") == 1, m.occurrences_of("archive") == 1)
-        {
+        let dump_count =
+            m.get_many::<usize>("dump").map(|m| m.len()).unwrap_or(0);
+        let archive_count =
+            m.get_many::<usize>("archive").map(|m| m.len()).unwrap_or(0);
+        match (dump_count == 1, archive_count == 1) {
```

Comment:
@epage feel absolutely zero obligation to respond, but this is one change where this API seems to be slightly more verbose. I don't know if that's expected or if I'm maybe doing something wrong here, if there's a more modern UI to make this happen, I'd love to upgrade this code!

Reply:
> we want to allow an explicitly set value (that is, via the command line) to win the conflict.

`occurrences_of` was an indirect way of checking if something was set on the command-line.  We now have `ArgMatches::value_source(id) -> Option<ValueSource>` which let's you explicitly check if the value came from the command-line or from the environment (#3844 updates the deprecation message).  `ValueSource` is also `PartialEq + Eq + PartialOrd + Ord`.  Unsure if a `match` is preferred or seeing which one has a higher priority source.

On the other hand, the approach this was replaced with just returns the number of values, regardless of where it came from.

btw I wonder if there is an ergonomic enough / not-too-terrible-document way of exposing `value_source` in the derive API.  Feel free to create an issue about it.  The only other piece of information I can think of that is in `ArgMatches` but not in the derive is arg group values.  We push all values that belong to arguments in groups to the group's id.  I do want to change that as it seems wasteful when I bet most people don't use it or don't even know it exists.  I think I'll make an `Action` to control it.

Reply:
Nice, this worked really well. Thank you!

> Unsure if a match is preferred or seeing which one has a higher priority source.

In this specific case, you can only have one of either of these, the more accurate check would be "not zero" rather than "is one" in some sense. The trick is mostly that folks will often want to provide these via environment variables for normal use, and then use the command line to override that env var when doing something special. So we don't want to make them actually exclusionary, because then users would have to go remove their normal environment to get things to work, which would be awkward.

I'm not sure how to expose it via the derive API either, to be honest. If I can think of anything, let me know.